### DISCUSSION TITLE
z.array().safeParse throwing error

### DISCUSSION THREAD

**Discussion by FearTheBeard95** [colinhacks/zod#3106] [Q&A] [answered]
Im getting an error while using `z.array` around a `z.object` schema for example 

```ts
const products = z.array(
  z.object({
    product_id: z.number(),
    name_en: z.string()
  })
);

products.safeParse([
  {
    product_id: 1,
    name_en: 'test'
  }
]);
```

cause the error

```
TypeError: effect.transform is not a function
    at ZodEffects._parse (types.js:3375:31)
    at ZodObject._parse (types.js:1957:29)
    at ZodObject._parseSync (types.js:170:25)
    at eval (types.js:1811:23)
    at Array.map (<anonymous>)
    at ZodArray._parse (types.js:1810:34)
    at ZodArray._parseSync (types.js:170:25)
    at ZodArray.safeParse (types.js:207:25)
    at eval (products.ts:722:63)
    at step (products.ts:146:17)
```

**Comment by JacobWeisenburger** [comment, marked as answer]
Everything is working as expected for me. 

```ts
const products = z.array( z.object( { product_id: z.number(), name_en: z.string() } ) )
const result = products.safeParse( [ { product_id: 1, name_en: 'test' } ] )
result.success && console.log( result.data )
// [
//     {
//         product_id: 1,
//         name_en: "test"
//     }
// ]
```

Please send a full reproducible code example.

**Reply by FearTheBeard95** [reply]
here's the link to the [sandbox](https://codesandbox.io/p/devbox/unruffled-fog-gjsx4v?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clr0c1e7700063b6hglwemmkl%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clr0c1e7700023b6h7r1dud5q%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clr0c1e7700033b6h046bqivb%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clr0c1e7700053b6hnyle5kbe%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clr0c1e7700023b6h7r1dud5q%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clr0c1e7600013b6hh35nltis%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fpublic%252Findex.html%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clr0c1e7700023b6h7r1dud5q%2522%252C%2522activeTabId%2522%253A%2522clr0c1e7600013b6hh35nltis%2522%257D%252C%2522clr0c1e7700053b6hnyle5kbe%2522%253A%257B%2522id%2522%253A%2522clr0c1e7700053b6hnyle5kbe%2522%252C%2522activeTabId%2522%253A%2522clr0c7703007j3b6hbfncj1r8%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522start%2522%252C%2522port%2522%253A3000%252C%2522id%2522%253A%2522clr0c7703007j3b6hbfncj1r8%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252F%2522%257D%255D%257D%252C%2522clr0c1e7700033b6h046bqivb%2522%253A%257B%2522id%2522%253A%2522clr0c1e7700033b6h046bqivb%2522%252C%2522activeTabId%2522%253A%2522clr0c7298005f3b6hjj5fmtc6%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522start%2522%252C%2522id%2522%253A%2522clr0c7298005f3b6hjj5fmtc6%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Afalse%252C%2522sidebarPanelSize%2522%253A15%257D)

It appears that I'm facing intermittent issues with my code. The data I'm attempting to validate sometimes works and sometimes doesn't. Interestingly, it runs successfully from one location but exhibits inconsistent behavior elsewhere.

I'm curious if there might be an issue with the implementation. Could you provide insights into potential problems or suggest areas to investigate?

**Reply by JacobWeisenburger** [reply]
I can't find the code you are talking about in your example, please narrow it down to just the problem code.
